pkg/build/generator/generator.go

```diff
@@ -312,6 +313,10 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	if build.Annotations == nil {
+		build.Annotations = make(map[string]string)
+	}
+	build.Annotations[buildapi.BuildNumberAnnotation] = strconv.Itoa(bc.Status.LastVersion)
```

Comment:
can you add a TODO in `generateBuildFromBuild` to figure out what to do with build numbers on clones of builds?